### PR TITLE
Removing indentation prop and moving it to the themes.

### PR DESCRIPTION
**Pull request by MayaKirova**
Related to #1408

@simeonoff or @desig9stein could you take a look at the sass files? I've added some very basic styles so that it does not look broken, but it can probably be styled better.

### CODE REVIEW

File: projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss [hunk 1]
```diff
@@ -693,6 +693,21 @@
...
+            min-width: #{$i*30}px;
+            flex-basis: #{$i*30}px;
+
+            .igx-button--icon {
+                min-width: 30px;
+                flex-basis: 30px;
+            }
+        }
+        %grid__group-indentation-padding-level-#{$i} {
+            padding-left: #{$i*30}px;
```
[the remark this answers is not in the document]
Comment (MayaKirova) [reply projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss]: This was originally  bound to the padding:
[style.padding-left]="padding"
So I just moved it as is to the theme.
If it makes more sense to be margin let me know and/or feel free to push the change directly to the branch.

File: projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss [hunk 2]
```diff
@@ -693,6 +693,21 @@
...
+    @for $i from 1 through 10 {
+        %grid__group-indentation-level-#{$i} {
+            min-width: #{$i*30}px;
+            flex-basis: #{$i*30}px;
+
+            .igx-button--icon {
+                min-width: 30px;
+                flex-basis: 30px;
+            }
+        }
```
Comment (desig9stein) [inline projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss]: The smallest grouping-indicator should be min-width:60px, flex-basis: 60px

Comment (desig9stein) [reply projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss]: Respectively the smallest group-indentation should be 60px

File: projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (simeonoff) [reply projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss]: After discussing with @desig9stein we decided to keep it as padding as the background is part of the element.

File: projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss (hunk 2 above)
Comment (desig9stein) [reply projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss]: I will test and do that change

Comment (MayaKirova) [reply projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss]: Ok. Thank you  :+1:

**Pull request merged**